### ISSUE TITLE
GDAL Translate error 4

### ISSUE THREAD

**Issue by aleksanh** [qgis/QGIS#45306]
### What is the bug or the crash?

Always get an error with GDAL in QGIS. 
this happens for other gdal generated commands to. win
windows 10... 

**QGIS version: 3.20.3-Odense
QGIS code revision: 495fbaec
Qt version: 5.15.2
Python version: 3.9.5
GDAL version: 3.3.1
GEOS version: 3.9.1-CAPI-1.14.2
PROJ version: Rel. 8.1.1, September 1st, 2021
PDAL version: 2.3.0 (git-version: eac774)
Processing algorithm…
Algorithm 'Clip raster by extent' starting…
Input parameters:**
`{ 'DATA_TYPE' : 0, 'EXTRA' : '', 'INPUT' : 'type=xyz&url=https://tile.openstreetmap.org/%7Bz%7D/%7Bx%7D/%7By%7D.png&zmax=19&zmin=0', 'NODATA' : None, 'OPTIONS' : '', 'OUTPUT' : 'C:/seatex/Data_and_files/demo_osterdam/Bridge_test/qimera/Osterdam_demo/Export/open_s_small.tif', 'OVERCRS' : False, 'PROJWIN' : '585648.547700000,586010.047700000,5697896.579100000,5698382.579100000 [EPSG:32631]' }`

```
GDAL command:
gdal_translate -projwin 471095.6177414938 6697740.467284465 471687.4290837575 6696950.645862748 -of GTiff "type=xyz&url=https://tile.openstreetmap.org/%7Bz%7D/%7Bx%7D/%7By%7D.png&zmax=19&zmin=0" C:/seatex/Data_and_files/demo_osterdam/Bridge_test/qimera/Osterdam_demo/Export/open_s_small.tif
```
**GDAL command output:
ERROR 4: `type=xyz&url=https://tile.openstreetmap.org/%7Bz%7D/%7Bx%7D/%7By%7D.png&zmax=19&zmin=0' does not exist in the file system, and is not recognized as a supported dataset name.
Process returned error code 1**
Execution completed in 0.07 seconds
**Results:
{'OUTPUT': 'C:/seatex/Data_and_files/demo_osterdam/Bridge_test/qimera/Osterdam_demo/Export/open_s_small.tif'}**

**Loading resulting layers
The following layers were not correctly generated.
• C:/seatex/Data_and_files/demo_osterdam/Bridge_test/qimera/Osterdam_demo/Export/open_s_small.tif
You can check the 'Log Messages Panel' in QGIS main window to find more information about the execution of the algorithm.**

### Steps to reproduce the issue

Run raster-extraction-clip raster by extent on google satelite layer or open street map. 

### Versions

3.20.3-odense

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by gioman** [comment]
> Run raster-extraction-clip raster by extent on google satelite layer or open street map.

@aleksanh I suspect that simply GDAL tools (and probably other in Processing) are not meant to handle that datasources.

**Comment by alexbruy** [comment]
@gioman you are right, most of the Processing tools does not support such datasources. Probably it should be turned into feature request.
